### PR TITLE
Clarifying unit tests

### PR DESCRIPTION
**Pull request by danbev**
Motivation:
Some tests are a little hard to follow with regards to the expectations
in the diff-sync specification. Adding some comments to the code will
hopefully help us in the future to understand what the test do. This is
especially true for the tests that deal with restoring the client backup
shadow.

Modifications:
Updated patchRevertToBackup adding comments and a few minor refactorings
for clarity.

Result:
Will hopefully be easier for us to understand the test and make updates
in the future.

JIRA: https://issues.jboss.org/browse/AEROGEAR-1585

### CODE REVIEW

**Comment by matzew** [comment]
:+1:

**Comment by danielpassos** [comment]
Landed at 48850d179314c8a930c7575ea29fb2ca3b8a2e99

**Comment by danielpassos** [comment]
:+1:

**Comment by danielpassos** [comment]
Landed at 5c06368f32dbe1262e678c47b0723cdc24596129. I'm sorry for my mistake.
